records_mover/db/driver.py

```diff
@@ -68,16 +68,14 @@ def set_grant_permissions_for_groups(self,
                                          db_engine: Optional[sqlalchemy.engine.Engine] = None
                                          ) -> None:
         db, db_conn, db_engine = check_db_conn_engine(db=db, db_conn=db_conn, db_engine=db_engine)
-        schema_and_table: str = quote_schema_and_table(None, schema_name, table,
-                                                       db_engine=self.db_engine)
         for perm_type in groups:
             groups_list = groups[perm_type]
             for group in groups_list:
-                group_name: str = quote_group_name(None, group, db_engine=self.db_engine)
                 if not perm_type.isalpha():
                     raise TypeError("Please make sure your permission types"
                                     " are an acceptable value.")
-                perms_sql = f'GRANT {perm_type} ON TABLE {schema_and_table} TO {group_name}'
+                table_obj = Table(table, MetaData(), schema=schema_name)
+                perms_sql = str(GrantPrivileges(perm_type, table_obj, group))
```

Comment:
should this be `text()` instead of `str()` or am i misunderstanding how you're supposed to you text()?

Reply:
need both because text expects a string or bytes-like object; here's the TypeError you get without converting first:
`TypeError: expected string or bytes-like object`